src/libcore/option.rs

```diff
@@ -295,10 +295,16 @@ impl<T> Option<T> {
     pub fn expect(self, msg: &str) -> T {
         match self {
             Some(val) => val,
-            None => panic!("{}", msg),
+            None => Self::expect_failed(msg),
         }
     }
 
+    #[inline(never)]
+    #[cold]
+    fn expect_failed(msg: &str) -> ! {
+        panic!("{}", msg)
+    }
```

Comment:
Could this be moved outside of `Option<T>`? That should mean it's not monomorphized and should be even smaller I believe